partiql-plan/src/main/java/org/partiql/plan/rel/RelWith.java

```diff
@@ -0,0 +1,105 @@
+package org.partiql.plan.rel;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.plan.Operand;
+import org.partiql.plan.OperatorVisitor;
+import org.partiql.plan.WithListElement;
+import org.partiql.plan.rex.Rex;
+
+import java.util.List;
+
+/**
+ * <p>
+ * <b>NOTE:</b> This is experimental and subject to change without prior notice!
+ * </p>
+ * <p>
+ * WITH relation expression. This is currently experimental and is currently missing some
+ * features like recursive CTEs.
+ */
+public abstract class RelWith extends RelBase {
```

Comment:
self-review: `WITH`'s representation in the partiql-plan is marked as experimental for now. We may make public API changes as we add the remaining SQL features related to WITH.